layers/+tools/lsp/funcs.el

```diff
@@ -30,3 +30,79 @@
                       :background (face-attribute 'highlight :background nil t)
                       :foreground (face-attribute 'default :foreground nil t))
   )
+
+(defun spacemacs/lsp-append-jump-handlers (mode)
+    ;;; Override
+  (let ((handler (intern (format "spacemacs-jump-handlers-%s" mode))))
+    (add-to-list handler 'lsp-ui-peek-find-definitions))
+  ;; The notion of 'spacemacs-reference-handlers' is the subject of this PR:
+  ;; https://github.com/syl20bnr/spacemacs/pull/9911
+  ;; Disabling for now...
+  ;; (let ((handler (intern (format "spacemacs-reference-handlers-%s" mode))))
+  ;;   (add-to-list handler 'lsp-ui-peek-find-references))
+  )
+
+(defun spacemacs/lsp-bind-keys-for-mode (mode)
+  "Define key bindings for the specific MODE."
+  (spacemacs/declare-prefix-for-mode mode "m=" "format")
+  (spacemacs/declare-prefix-for-mode mode "mg" "goto")
+  (spacemacs/declare-prefix-for-mode mode "ml" "lsp")
+  (spacemacs/declare-prefix-for-mode mode "mr" "refactor")
+  (spacemacs/declare-prefix-for-mode mode "mt" "toggle")
+
+  (spacemacs/set-leader-keys-for-major-mode mode
+    ;;Format
+    "==" #'spacemacs/lsp-format-buffer
+    ;;goto
+    "gi" #'imenu
+    "gs" #'lsp-ui-find-workspace-symbol
+    "gf" #'spacemacs/lsp-ffap-no-prompt
```

Comment:
ffap does not use LSP.

Reply:
Also removed. Originally included that a binding like this was appropriate in a general purpose lsp layer, but maybe there's a better idiomatic equivalent, i.e an lsp method that would jump to a header file when invoked on a c source #include, or the appropriate ruby source file when invoked on a  'require...' statement etc.?